services/movie_session.py

```diff
@@ -0,0 +1,53 @@
+from typing import List
+
+from db.models import Movie, CinemaHall, MovieSession
+import datetime
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> None:
+    work_movie = Movie.objects.get(id=movie_id)
+    work_cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+    new_movie_session = MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie=work_movie,
+        cinema_hall=work_cinema_hall
+    )
+    return new_movie_session
+
+
+def get_movies_sessions(session_date: str = None) -> List[MovieSession]:
+    queryset = MovieSession.objects.all()
+    if session_date is not None:
+        date_format = "%Y-%m-%d"
+        work_date = datetime.datetime.strptime(session_date, date_format)
+        queryset = queryset.filter(show_time__date=work_date)
+
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: MovieSession) -> Movie:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None

```

Comment:
use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contains string elements and can be None has typehint Optional[list[str]]